source/plugin/messagechannels.rst

```diff
@@ -0,0 +1,227 @@
+
+        @Override
+        public Collection<MessageReceiver> getMembers() {
+            return Collections.emptyList();
+        }
+    }
+
+
+Note that we do not wish to define any additional recipients, so we return an empty collection in the ``getMembers``
+method.
+
+Thanks to the capabilities of combined ``MessageChannel``\ s, we can combine our newly created ``AdminMessageChannel``
+with any other ``MessageChannel``. Now if a player joins with the ``myplugin.admin`` permission, we will obtain the
+``MessageChannel`` his messages are sent to, combine it with an ``AdminMessageChannel`` and set the combined channel
+back onto the player. That way all his messages are sent to everyone specified in the original channel, but due to the
+addition of the ``AdminMessageChannel``, a red ``[Admin]`` tag will be prefixed.
+
+.. code-block:: java
+
+    import org.spongepowered.api.entity.living.player.Player;
+    import org.spongepowered.api.event.Listener;
+    import org.spongepowered.api.event.network.ClientConnectionEvent;
+
+    private AdminMessageChannel adminChannel = new AdminMessageChannel();
+
+    @Listener
+    public void onClientConnectionJoin(ClientConnectionEvent.Join event) {
+        Player player = event.getTargetEntity();
+        if(player.hasPermission("myplugin.admin")) {
+            MessageChannel originalChannel = event.getOriginalChannel();
+            MessageChannel newChannel = MessageChannel.combined(originalChannel,
+                adminChannel);
+            player.setMessageChannel(newChannel);
+        }
+    }
+
+.. note::
+
+    When combining multiple ``MessageChannel``\ s defining different message transformations, the ``Text`` will be
+    transformed in the order that the ``MessageChannel``\ s are passed in to the
+    ``MessageChannel#combined(MessageChannel... channels)`` method. Note that any transformations resulting in an
+    empty ``Optional`` will be ignored unless performed by the last channel in the chain.
+
+Mutable Message Channels
+========================
+
+A ``MutableMessageChannel`` contains methods for changing who may receive the messages sent through our channel. As
+such, we must implement methods for performing actions that modify our members. To do this, we simply will create a
+class named ``MutableAdminMessageChannel`` that will implement a ``MutableMessageChannel``.
+
+.. code-block:: java
+
+    import java.util.HashSet;
+    import java.util.Set;
+
+    import org.spongepowered.api.text.channel.MutableMessageChannel;
+
+    public class MutableAdminMessageChannel implements MutableMessageChannel {
+
+        private Set<MessageReceiver> members;
+
+        public MutableAdminMessageChannel() {
+            this(new HashSet<>());
+        }
+
+        public MutableAdminMessageChannel(Set<MessageReceiver> members) {
+            this.members = members;
+        }
+
+        @Override
+        public Collection<MessageReceiver> getMembers() {
+            return members;
```

Comment:
Should return an Immutable View of the set so that users may only be added / removed through the appropriate methods. (see `Collections#unmodifiableSet`) or guavas `ImmutableSet.of()``